Source/Dafny/Resolver.cs

```diff
@@ -12244,7 +12245,21 @@ class LazyString_OnTypeEquals
                   } else {
                     // pick the supertype "mbr.EnclosingClass" of "cl"
                     Contract.Assert(mbr.EnclosingClass is TraitDecl);  // a proper supertype of a ClassDecl must be a TraitDecl
-                    var proxyTypeArgs = mbr.EnclosingClass.TypeArgs.ConvertAll(_ => (Type)new InferredTypeProxy());
+                    var typeMapping = cl.ParentFormalTypeParametersToActuals;
+                    List<Type> proxyTypeArgs = null;
+                    foreach (Type tt in cl.ParentTraits) {
```

Comment:
This will only iterate through the immediate parent traits. Don't you also want transitive parent traits? To do that, use `TraitAncestors()`.

Reply:
Changed